sklearn/model_selection/_search.py

```diff
@@ -384,7 +384,11 @@ def __repr__(self):
 
 class BaseSearchCV(six.with_metaclass(ABCMeta, BaseEstimator,
                                       MetaEstimatorMixin)):
-    """Base class for hyper parameter search with cross-validation."""
+    """Abstract base class for hyper parameter search with cross-validation.
+
+    Implementers should provide ``__init__`` and either ``_get_param_iterator``
```

Comment:
for my education: why do we need to provide `__init__` if it is empty (like in the test)? Doesn't python call the base `__init__` automatically?